backend/typescript/services/implementations/authService.ts

```diff
@@ -265,6 +270,39 @@ class AuthService implements IAuthService {
       return false;
     }
   }
+
+  async getFirebaseUserByUid(
+    uid: string,
+  ): Promise<firebaseAdmin.auth.UserRecord> {
+    let firebaseUser: firebaseAdmin.auth.UserRecord;
+    try {
+      firebaseUser = await firebaseAdmin.auth().getUser(uid);
+      if (!firebaseUser) throw new Error(`No user found with uid: ${uid}`);
+      return firebaseUser;
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to get firebase user by uid: ${uid}. Reason = ${getErrorMessage(
+          error,
+        )}`,
+      );
+      throw error;
+    }
+  }
+
+  async verifyEmail(uid: string): Promise<void> {
```

Comment:
Does this function handle the actual verification or simply setting `emailVerified = true`? If the latter, we should prolly change its name to something like `markVerified` to remove ambiguity

Reply:
Good point. I'll rename it to markVerified